options.js

```diff
@@ -1,42 +1,65 @@
-chrome.storage.local.get(
-    ["options", "exclusionArray"],
-    ({
-        options = {
-            moveTabs: true,
-            effectWindows: false,
-            effectTabGroups: false,
-        },
-        exclusionArray = [],
-    }) => {
-        const { moveTabs, effectWindows, effectTabGroups } = options;
+
+/**
+ * Read from local storage in async instead of using callbacks
+ * Pass in a single string key or an array of keys to retrieve multiple values
+ * 
+ * @param {string|array<string>} key 
+ * @returns {object}
+ */
+const getLocalStorageKey = key => {
+    let storageKey = key;
+    if (typeof storageKey !== 'array') storageKey = [storageKey];
+
+    return new Promise((resolve, reject) => {
+        try {
+            chrome.storage.local.get(key, resolve)
+        } catch (err) {
+            reject(err);
+        }
+    });
+}
+
+/**
+ * Set an object in local storage to the given value
+ * Local Stoarge objects key is based on the Object key passed in
+ * @param {Object} val 
+ */
+const setLocalStorageValue = val => chrome.storage.local.set(val);
+
+/**
+ * Upsert into options local storage namespace
+ * Local Stoarge objects key is based on the Object key passed in
+ * @param {Object} val 
+ */
+const updateOptions = async (updatedValues) => {
+    const { options } = await getLocalStorageKey();
+    const updatedOptions = {
+        ...options,
+        ...updatedValues,
+    }
+    setLocalStorageValue({ options: updatedOptions });
+}
+
+getLocalStorageKey('options')
+    .then(({ options }) => {
+        const { moveTabs, effectWindows, effectTabGroups, exclusions } = options;
         document.getElementById("moveTabs").checked = moveTabs;
         document.getElementById("effectWindows").checked = effectWindows;
         document.getElementById("effectTabGroups").checked = effectTabGroups;
-        createHTMLList(exclusionArray);
-        chrome.storage.local.set({
-            options: {
-                moveTabs,
-                effectWindows,
-                effectTabGroups,
-            },
-            exclusionArray,
-        });
-    }
-);
+        createHTMLList(exclusions);
+    });
+
 
 const submit = document.getElementById("submit");
 const exclusionBox = document.getElementById("exclusionArray");
 const exclusionList = document.getElementById("exclusions");
 
-function isPropertyEnabled(id) {
+function isOptionEnabled(id) {
     return document.getElementById(id).checked;
 }
 
-function createHTMLList(exclusionArray) {
-    exclusionArray.forEach((item) => {
-        addToHTMLList(item);
-
-    });
+function createHTMLList(exclusions) {
```

Comment:
In the future, you could make this even more broad and take in the actual list as a parameter.

Imagine you build a history list. You could reuse this exact function as is. As-is though its fine because we don't have a usecase for another list. There are some philosophies on this. Check out https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it